.github/workflows/test-os.yml

```diff
@@ -162,7 +168,7 @@ jobs:
         if: always()
         uses: actions/upload-artifact@v4
         with:
-          name: test-reports-${{ env.TEST_REPORT_NAME }}
+          name: test-reports-${{ matrix.os }}-${{ env.TEST_REPORT_NAME }}
```

Comment:
may be the `matrix.os` should come at the end of the name, but no strong preference.